crates/sui-indexer/src/apis/indexer_api.rs

```diff
@@ -194,10 +181,13 @@ impl<S: IndexerStore> IndexerApi<S> {
             }
         }?;
 
-        // digests here are of size (limit + 1), where the last one is the cursor for the next page
-        let mut tx_digests = digests_from_db
-            .iter()
-            .map(|digest| {
+        let has_next_page = tx_vec_from_db.len() > limit;
```

Comment:
question: what's the reason for creating this limit in the api instead of when we make the query? Not related to this PR but the question came up while reviewing haha

without cursor:
```
SELECT * 
FROM your_table 
LIMIT 1000 OFFSET 1000;
```

with cursor:
```
SELECT * 
FROM your_table 
WHERE id > last_element_id 
ORDER BY id 
LIMIT 1000;
```

Reply:
oh ig its cuz we wuldn't know if there are more pages without paginating ...
maybe we can in postgres make limit+1?

Reply:
probs a premature optimization but I think these transactions pile up quickly

Reply:
not sure if I follow entirely, but overall the `pagination logic` should stay in `*_api.rs` including fetching an extra item for has_next_page; while query just handles query with filters (including cursor) and limit.

now some logic / types are still coupled, I also want to further split it when I have time lol